scipy/stats/_continued_fraction.py

```diff
@@ -287,36 +293,7 @@ def b(n, *args, b=b):
     def func(n, *args):
         return xp.stack((a(n, *args), b(n, *args)), axis=-1)
 
-    # Initialization
-    # The EIM framework was written with only one callable in mind. Again,
-    # rather than complicating the framework, we call its `initialize` function
-    # on each callable to get the shape and dtype, then we broadcast these
-    # shapes, compute the result dtype, and broadcast/promote the zeroth terms
-    # and `*args` to this shape/dtype.
-
-    # `float32` here avoids influencing precision of resulting float type
-    # patch up promotion: in numpy (int64, float32) -> float64, while in torch
-    # (int64, float32) -> float32 irrespective of the default_dtype.
-    dt = {'dtype': None
-          if is_torch(xp) and xp_default_dtype(xp) == xp.float64
-          else xp.float32}
-    zero = xp.asarray(0, **dt)
-
-    temp = eim._initialize(a, (zero,), args, complex_ok=True)
-    _, _, fs_a, _, shape_a, dtype_a, xp_a = temp
-    temp = eim._initialize(b, (zero,), args, complex_ok=True)
-    _, _, fs_b, _, shape_b, dtype_b, xp_b = temp
-
-    xp = array_namespace(fs_a[0], fs_b[0], *args)
-
-    shape = np.broadcast_shapes(shape_a, shape_b)  # OK to use NumPy on tuples
-    dtype = xp.result_type(dtype_a, dtype_b)
-    an = xp.astype(xp_ravel(xp.broadcast_to(xp.reshape(fs_a[0], shape_a), shape)), dtype)  # noqa: E501
-    bn = xp.astype(xp_ravel(xp.broadcast_to(xp.reshape(fs_b[0], shape_b), shape)), dtype)  # noqa: E501
-    args = [xp.astype(xp_ravel(xp.broadcast_to(arg, shape)), dtype) for arg in args]
-
-    status = xp.full_like(an, xp.asarray(eim._EINPROGRESS),
-                          dtype=xp.int32)  # in progress
```

Comment:
I confess I feel out of my depth here and I have no idea what this PR functionally changes in this module, if anything.

Reply:
No problem.